File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //
    string text = get_string("Text: ");

    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float L = 100 * letters / words;

    float S = 100 * sentences / words;

    // index = 0.0588 * L - 0.296 * S - 15.8
    float index_ = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(index_);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index > 0)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Before Grade 1\n");
    }


}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")


letters = 0
for i in text:
    if i.isalpha():
        letters += 1

words = 1
for j in text:
    if j == ' ':
        words += 1

sentences = 0
for k in text:
    if k == '.' or k == '!' or k == '?':
        sentences += 1


L = float(100 * letters / words)
S = float(100 * sentences / words)

# 0.0588 * L - 0.296 * S - 15.8

index = float(0.0588 * L - 0.296 * S - 15.8)

grade = round(index)

if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print("Grade: ", grade)